include/Token.h

```diff
@@ -197,6 +197,7 @@ enum class KeywordVersion : uint8_t {
 };
 
 const StringTable<TokenKind>* getKeywordTable(KeywordVersion version);
+bool isKeywordOrIdentifier(TokenKind kind);
```

Comment:
I think the meaning of the return value would be clearer if this were just, say "isKeyword" or something -- having the or is a bit confusing.

Also, maybe add a comment that this actual checks all keywords, so should only be used conservatively where you are ok letting them be interhchangeable (like Macro arguments) it will produce incorrect results if you really need to know whether or not something is a keyword and the keyword table is not the default due to the begin_keywords directive.